### PR TITLE
[FeG] Use generated mock interfaces for testing

### PR DESCRIPTION
**Pull request by themarwhal**
## Summary
Noticed that we manually write mock interfaces for testing when we can use the generated ones. Migrated session_controller_test.go to use the generated files, and added `gen.go` files to ensure they stay up-to-date.
From now on, running `make gen` at the top level should generate all these mock dependencies

(https://github.com/vektra/mockery) <- mock generator

## Test Plan
make precommit FeG

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: feg/gateway/services/session_proxy/servicers/session_controller_test.go [hunk 1]
```diff
@@ -238,11 +158,10 @@ func TestStartSessionGyFail(t *testing.T) {
 		}
 	}).Once()
 
-	mockPolicyDb.On("GetChargingKeysForRules", mock.Anything, mock.Anything).Return(
+	mockPolicyDBClient.On("GetChargingKeysForRules", mock.Anything, mock.Anything).Return(
 		[]policydb.ChargingKey{{RatingGroup: 1}}, nil).Once()
 	// no omnipresent rules
-	mockPolicyDb.On("GetOmnipresentRules").Return([]string{}, []string{}).Once()
-	mockPolicyDb.On("GetRuleIDsForBaseNames", mock.Anything).Return([]string{}).Once()
```
Comment (uri200) [inline feg/gateway/services/session_proxy/servicers/session_controller_test.go:245]: why this has disappeared?

**Review by uri200 — COMMENTED** [review]
aren't we losing the flexibility to add some kind of custom logic to the mock in the future?

**Comment by themarwhal** [comment]
@uri200 The code I removed did not have any custom logic :) It's not really stopping us from creating custom mock classes in the future, but right now if I change one interface I find myself modifying several files, some generated some not.

File: feg/gateway/services/session_proxy/servicers/session_controller_test.go (hunk 1 above)
Comment (themarwhal) [reply feg/gateway/services/session_proxy/servicers/session_controller_test.go:245]: I added this `mockPolicyDBClient.AssertExpectations(t)` down at line 188 and this call was never called. We were never asserting on the expectations before.

**Review by uri200 — APPROVED** [review]

**Pull request merged**